### PR TITLE
global sort: support setting cloud storage uri variable

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/45719

Problem Summary:

### What is changed and how it works?
support setting global sort uri variable

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

```sql
mysql> set @@global.tidb_global_sort_uri="s3://tiflow-test?access-key=testid&secret-access-key=testkey8&endpoint=http://127.0.0.1:5000";
Query OK, 0 rows affected (0.05 sec)
mysql> select @@global.tidb_global_sort_uri;
+-------------------------------+
| @@global.tidb_global_sort_uri |
+-------------------------------+
| s3://tiflow-test/             |
+-------------------------------+
1 row in set (3.69 sec)
```

Side effects

- None

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
support setting tidb_global_sort_uri variable
```

### CODE REVIEW

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -2227,6 +2227,17 @@ var defaultSysVars = []*SysVar{
 		DDLDiskQuota.Store(TidbOptUint64(val, DefTiDBDDLDiskQuota))
 		return nil
 	}},
+	{Scope: ScopeSession, Name: TiDBGlobalSortURI, Value: "", Type: TypeStr, Validation: func(sv *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
+		if err := ValidGlobalSortURI(normalizedValue); err != nil {
+			return "", err
+		}
+		return normalizedValue, nil
+	}, GetSession: func(sv *SessionVars) (string, error) {
+		return sv.GlobalSortURI, nil
```
Comment (Benjamin2037) [inline sessionctx/variable/sysvar.go:2236]: Do we need add mask function to hide ak sk information？

**Comment by lichunzhu** [comment]
@tangenta  @Benjamin2037 PTAL again. I add a manual test in PR description

File: sessionctx/variable/tidb_vars.go [hunk 2]
```diff
@@ -982,6 +982,8 @@ const (
 	TiDBDDLEnableFastReorg = "tidb_ddl_enable_fast_reorg"
 	// TiDBDDLDiskQuota used to set disk quota for lightning add index.
 	TiDBDDLDiskQuota = "tidb_ddl_disk_quota"
+	// TiDBGlobalSortURI used to set a global sort uri for ddl add index and import into.
+	TiDBGlobalSortURI = "tidb_global_sort_uri"
```
Comment (tangenta) [inline sessionctx/variable/tidb_vars.go:986]: How about renaming it to `@@tidb_external_storage_uri`? @Frank945946 @easonn7

**Review by tangenta — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: sessionctx/variable/sysvar.go [hunk 3]
```diff
@@ -2256,6 +2256,20 @@ var defaultSysVars = []*SysVar{
 		DDLDiskQuota.Store(TidbOptUint64(val, DefTiDBDDLDiskQuota))
 		return nil
 	}},
+	// can't assign validate function here. Because validation function will run after GetGlobal function
+	{Scope: ScopeGlobal, Name: TiDBCloudStorageURI, Value: "", Type: TypeStr, GetGlobal: func(ctx context.Context, sv *SessionVars) (string, error) {
+		backend, err := ParseCloudStorageURI(ctx, CloudStorageURI.Load(), false)
+		if err != nil {
+			return "", err
+		}
+		return backend.URI(), nil
```
Comment (D3Hunter) [inline sessionctx/variable/sysvar.go:2265]: why not return a redacted value?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**